go/tools/builders/nogo_vet.go

```diff
@@ -111,16 +110,27 @@ func buildVetcfgFile(packageFile, importMap map[string]string, stdImports, files
 		vcfg.Standard[imp] = true
 	}
 
-	js, err := json.MarshalIndent(vcfg, "", "\t")
+	// GRIPE: vet checks whether its first positional argument has the suffix
+	// "vet.cfg", rather than accepting the file as an option.
+	vcfgFile, err := ioutil.TempFile("", "nogo-*-vet.cfg")
 	if err != nil {
-		return "", fmt.Errorf("internal error marshaling vet config: %v", err)
-	}
-	js = append(js, '\n')
-	vcfgFile := filepath.Join(os.TempDir(), "vet.cfg")
-	if err := ioutil.WriteFile(vcfgFile, js, 0666); err != nil {
 		return "", err
 	}
-	return vcfgFile, nil
+	vcfgPath := vcfgFile.Name() // vcfgPath may be "" when returning
+	defer func() {
+		os.Remove(vcfgPath)
```

Comment:
Potentially stupid question: is this deferred Remove necessary or problematic when the calling function (nogo_main.run) also defers the removal of this file?

Reply:
The caller will only be responsible for removing the file if the call succeeds. The call could fail in a number of ways, including not being able to create the file, and the caller shouldn't have to worry about that. If the file is successfully created though, then it's the caller's responsibility.